src/library_fsfs.js

```diff
@@ -0,0 +1,225 @@
+          mode: handle.kind === 'file' ? FSFS.FILE_MODE : FSFS.DIR_MODE,
+        };
+      }
+
+      return { type: 'remote', entries, handles };
+    },
+    loadLocalEntry: (path) => {
+      const lookup = FS.lookupPath(path);
+      const node = lookup.node;
+      const stat = FS.stat(path);      
+
+      if (FS.isDir(stat.mode)) {
+        return { 'timestamp': stat.mtime, 'mode': stat.mode };
+      } else if (FS.isFile(stat.mode)) {
+        node.contents = MEMFS.getFileDataAsTypedArray(node);
+        return { timestamp: stat.mtime, mode: stat.mode, contents: node.contents };
+      } else {
+        throw new Error('node type not supported');
+      }
+    },
+    storeLocalEntry: (path, entry) => {
+      if (FS.isDir(entry['mode'])) {
+        FS.mkdirTree(path, entry['mode']);
+      } else if (FS.isFile(entry['mode'])) {
+        FS.writeFile(path, entry['contents'], { canOwn: true });
+      } else {
+        throw new Error('node type not supported');
+      }
+
+      FS.chmod(path, entry['mode']);
+      FS.utime(path, entry['timestamp'], entry['timestamp']);
+    },
+    removeLocalEntry: (path) => {
+      var stat = FS.stat(path);
+
+      if (FS.isDir(stat.mode)) {
+        FS.rmdir(path);
+      } else if (FS.isFile(stat.mode)) {
+        FS.unlink(path);
+      }
+    },
+    loadRemoteEntry: async (handle) => {
+      if (handle.kind === 'file') {
+        const file = await handle.getFile();
+        return {
+          contents: new Uint8Array(await file.arrayBuffer()),
+          mode: FSFS.FILE_MODE,
+          timestamp: file.lastModifiedDate,
+        };
+      } else if (handle.kind === 'directory') {
+        return {
+          mode: FSFS.DIR_MODE,
+          timestamp: new Date(),
+        };
+      } else {
+        throw new Error('unknown kind: ' + handle.kind);
+      }
+    },
+    storeRemoteEntry: async (handles, path, entry) => {
+      const parentDirHandle = handles.get(PATH.dirname(path));
+      const handle = FS.isFile(entry.mode) ?
+        await parentDirHandle.getFileHandle(PATH.basename(path), {create: true}) :
+        await parentDirHandle.getDirectoryHandle(PATH.basename(path), {create: true});
+      if (handle.kind === 'file') {
+        const writable = await handle.createWritable();
+        await writable.write(entry.contents);
+        await writable.close();
+      }
+      handles.set(path, handle);
+    },
+    removeRemoteEntry: async (handles, path) => {
+      const parentDirHandle = handles.get(PATH.dirname(path));
+      await parentDirHandle.removeEntry(PATH.basename(path));
+      handles.delete(path);
+    },
+    reconcile: async (mount, src, dst) => {
+      let total = 0;
+
+      const create = [];
+      Object.keys(src.entries).forEach(function (key) {
+        const e = src.entries[key];
+        const e2 = dst.entries[key];
+        if (!e2 || (FS.isFile(e.mode) && e['timestamp'].getTime() > e2['timestamp'].getTime())) {
```

Comment:
This line is peculiar, especially when you note that the idb interface does a simple equality check. The problem here is the the FS API does not allow you to set the mtime directly, it's just whatever the time was when it was written (as expected). So the modification time when written to disk will have to be ~syncfs time, which is strictly greater than the emscripten FS's mtime.

There's probably a race condition here, perhaps "syncfs" should be wrapped with some promise mechanism that prevents concurrent syncs? I'm not sure–I've ignored it because it can be avoided entirely if you just don't do concurrent syncs.